### ISSUE TITLE
Compatible with User Images?

### ISSUE THREAD

**Issue by callaginn** [radarseven/Craft-Twig-ImageBase64#3]
I might be using this plugin wrong, but is this compatible with user images? I'm trying to base64 encode Craft CMS user photos for some staff pages I've set up and base64 encode them.

I'm grabbing the URL segment to get the username, so that I can get the photoUrl.
```
{% set staff = craft.users.username( craft.request.getSegment(2) ).first() %}
```

However, it returns only a value of NULL whenever I run either of these commands. Is there something I'm doing wrong?
```
{{ image64(staff.photoUrl) }}
{{ staff.photoUrl|image64 }}
```

**Comment by radarseven** [comment]
Hey @callaginn, 

That unfortunately won't work with the plugin in it's current state as an `AssetFileModel` instance is required as the first parameter. Unfortunately, with Craft 2, the user photo isn't a native `Asset`, however, this will change with Craft 3.

I've marked this as an enhancement. I'll take a look at adding the ability to pass a `URL` or an `Asset` as a parameter.

Thanks,
Michael

**Comment by callaginn** [comment]
Thanks, that explanation is helpful. And it'd be awesome if you could pass a URL. I've been beating my head against the wall trying to figure out why it didn't work with user photos. Working on my first Craft site.

**Comment by radarseven** [comment]
@callaginn Try the `feature/url-parameter` branch which adds support for passing in an Asset URL. This will allow you to use it with the user's profile photo. Not sure you're intended use case, but given your example, you can now do:

    {% set staff = craft.users.username( craft.request.getSegment(2) ).first() %}

    {{ image64(staff.photoUrl) }}
    
    <img src="{{ image64(staff.photoUrl, true) }}" alt="{{ staff.fullName }}">


**Note**: Not extensively tested, but working on my end.

**Comment by callaginn** [comment]
@radarseven I copied the files from that branch, installed your plugin, and cleared my Craft caches, but no luck. Here's what I'm seeing when I use either my original code and the original branch or the currentUser code / new branch.
![screen shot 2017-03-21 at 1 29 55 pm](https://cloud.githubusercontent.com/assets/2430211/24165005/4ad926a4-0e3d-11e7-8fad-dc4687deec06.png)

I might need to explain my setup more, since I think we're thinking of two different use cases.

### User Profile Setup

I'm using standard Craft CMS User Accounts for content editors. I'm allowing these editors to edit content / set up personal staff pages, which will be viewable to the public. Since there is no currentUser, I'm pulling usernames from the URL to access user properties.
`{% set staff = craft.users.username( craft.request.getSegment(2) ).first() %}`

So if I'm correct, currentUser won't work...since I want every user profile to be viewable by the public.

### Why I'm using your plugin

I've got email signature and vCard templates set up as well. For instance, I'm inserting a URL like www.sitename.com/vcard/username into everyone's email signatures. Trying to use your plugin to base64 embed user images into the vCard template, which is set up using Craft routing.

### Simple vCard Template
```
{% set staff = craft.users.username( craft.request.getSegment(2) ).first() %}

BEGIN:VCARD
    N:{{ staff.lastName }};{{ staff.firstName }};
    {% for address in company.address %}
    ADR;INTL;PARCEL;WORK:;;{{ address.street }};{{ address.city }};{{ address.state }};{{ address.zip }};
    {% endfor %}
    EMAIL;INTERNET:{{ staff.email }}
    ORG:{{ siteName }};
    TITLE:{{ staff.jobTitle }}
    URL;WORK:{{ siteUrl }}
END:VCARD
```

**Comment by radarseven** [comment]
@callaginn Makes perfect sense. Your use case, creating a `VCARD`, is exactly why I created the plugin in the first place! 

I see the issue; it's the way I'm handling the `URL`. I'll take another stab at it and let you know when I have something for you to try.

**Comment by callaginn** [comment]
Yep, and I wanted something that'd be lightweight and versatile as well. For example, it'd be cool if it could create/pull gravatar urls as well. I think I remember that being related to base64? Not a priority though.

By the way, just finished my vCard signature. Everything's working now, besides images. Here's what I'm thinking. And if anyone else is reading this...you'll need to set up custom Craft fields for some of this : )

```
{% set staff = craft.users.username( craft.request.getSegment(2) ).first() %}

BEGIN:VCARD
VERSION:3.0

N:{{ staff.lastName }};{{ staff.firstName }};
FN:{{ staff.fullName }}
ORG:{{ siteName }};
TITLE:{{ staff.jobTitle }}
PHOTO;TYPE=JPEG;VALUE=URI:{{ image64(staff.photoUrl, true) }}
EMAIL:{{ staff.email }}
TEL;TYPE=WORK,VOICE:{{ staff.phone }}
URL:{{ siteUrl }}
{% for address in company.address %}
ADR;TYPE=WORK,PREF:;;{{ address.street }};{{ address.city }};{{ address.state }};{{ address.zip }};
{% endfor %}
SOURCE:{{ siteUrl }}vcard/{{ staff }}
UID:{{ siteName | replace(' ', '-') | lower }}_{{ staff.id }}
REV:{{ staff.dateUpdated }}

END:VCARD
```
